rust/etw_helpers/src/sessions.rs

```diff
@@ -0,0 +1,366 @@
+use std::{borrow::Cow, mem::ManuallyDrop};
+
+use windows::{core::PCWSTR, Win32::System::Diagnostics::Etw::*};
+
+#[repr(C)]
+pub struct EventTraceProperties {
+    props: EVENT_TRACE_PROPERTIES,
+    session_name: [u16; 1024],
+    file_name: [u16; 1024],
+}
+
+impl EventTraceProperties {
+    fn empty_for_query() -> EventTraceProperties {
+        unsafe {
+            let mut props: EventTraceProperties = core::mem::zeroed();
+            props.props.Wnode.BufferSize = core::mem::size_of::<Self>() as u32;
+            props.props.Wnode.Flags = WNODE_FLAG_TRACED_GUID;
+
+            props.props.LoggerNameOffset = core::mem::size_of::<EVENT_TRACE_PROPERTIES>() as u32;
+            props.props.LogFileNameOffset =
+                core::mem::size_of::<EVENT_TRACE_PROPERTIES>() as u32 + 1024;
+
+            props
+        }
+    }
+
+    pub fn set_session_name(&mut self, session_name: &str) -> &mut Self {
+        if !session_name.is_empty() {
```

Comment:
If session_name is empty, we should clear the existing name, so it shouldn't be a special case.

If session_name.len() >= 1024, we should error out.

No need to check i >= 1024 in the loop. Check len() ahead of time.